### PR TITLE
sol added to leetcode problem no 2200

### LINKED ISSUE #2650: [Feature Request]: add lc solution to problem no 2200
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

I would like to add lc solution to problem no 2200

### Use Case

solution addition

### Benefits

_No response_

### Add ScreenShots

_No response_

### Priority

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by AmruthaPariprolu**
## Related Issue

Fix #2650

## Description

[Please include a brief description of the changes or features added]

## Type of PR

- [ ] Bug fix
- [X ] Feature enhancement
- [ ] Documentation update
- [ ] Security enhancement
- [ ] Other (specify): _______________

## Screenshots / Videos (if applicable)

[Attach any relevant screenshots or videos demonstrating the changes]

## Checklist
- [X ] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have followed the code style guidelines of this project.
- [x] I have checked for any existing open issues that my pull request may address.
- [x] I have ensured that my changes do not break any existing functionality.
- [x] Each contributor is allowed to create a maximum of 4 issues per day. This helps us manage and address issues efficiently.
- [x] I have read the resources for guidance listed below.
- [x] I have followed security best practices in my code changes.

## Additional Context

[Include any additional information or context that might be helpful for reviewers.]

## Resources for Guidance

Please read the following resources before submitting your contribution:

- [ ] [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [ ] [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [ ] [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [ ] [Markdown Guide](https://www.markdownguide.org/)
- [ ] [MDX Documentation](https://mdxjs.com/docs/)
- [ ] [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- [ ] [Tabs: Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [ ] [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [ ] [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex) 

## Contribution Guidelines

Thank you for considering contributing to our project! To ensure smooth collaboration and effective contribution management, please adhere to the following guidelines:

### Issue Creation

1. **Limit on Issues:**
   - Each contributor is allowed to create a maximum of **4 issues per day**. This helps us manage and address issues efficiently.

### Contribution Levels

2. **Basic Contributions:**
   - This project is primarily focused on documentation. Most of the setup has been completed, so contributors will generally need to work on basic code tasks, such as writing tests.
   - For these tasks, issues will be assigned the **level1** label.

3. **Acknowledging Hard Work:**
   - If a contributor puts in significant effort on a task, the issue will be upgraded to **level2**. This is our way of recognizing and appreciating extra effort.

4. **Feature Additions and Component Work:**
   - Contributors working on new features or components using JSX/TSX will be assigned a level based on the complexity and quality of their work.
   - The more complex and valuable the contribution, the higher the level assigned.

### Level Definitions

- **Level1:**
  - Tasks are straightforward, such as fixing minor bugs, writing tests, or making simple documentation updates.
- **Level2:**
  - Tasks require more effort, such as addressing complex bugs, improving existing features, or making substantial documentation improvements.
- **Level3:**
  - Tasks are highly complex and involve significant new feature development, major refactoring, or extensive contributions to the project’s core components.

We look forward to your contributions and appreciate your effort in helping us improve the project!

### CODE REVIEW

File: dsa-solutions/lc-solutions/2200-2299/2200-Find All K-Distant Indices in an Array.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+  - Arrays
+  - C++
+  - Java
+  - Python
+description: "This document provides a solution where we will be returning an array in increasing order , satisfying the condition given"
+---
+
+## Problem
+
+You are given a 0-indexed integer array nums and two integers key and k. A k-distant index is an index i of nums for which there exists at least one index j such that |i - j| <= k and nums[j] == key.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/2200-2299/2200-Find All K-Distant Indices in an Array.md:15]: replace  `<` to `&lt;`

File: dsa-solutions/lc-solutions/2200-2299/2200-Find All K-Distant Indices in an Array.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+### Examples
+
+**Example 1:**
+
+Input: nums = [3,4,9,1,3,9,5], key = 9, k = 1
+Output: [1,2,3,4,5,6]
+Explanation: Here, nums[2] == key and nums[5] == key.
+- For index 0, |0 - 2| > k and |0 - 5| > k, so there is no j where |0 - j| <= k and nums[j] == key. Thus, 0 is not a k-distant index.
+- For index 1, |1 - 2| <= k and nums[2] == key, so 1 is a k-distant index.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/2200-2299/2200-Find All K-Distant Indices in an Array.md:27]: replace  `<` to `&lt;`

File: dsa-solutions/lc-solutions/2200-2299/2200-Find All K-Distant Indices in an Array.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+Return a list of all k-distant indices sorted in increasing order.
+
+### Examples
+
+**Example 1:**
+
+Input: nums = [3,4,9,1,3,9,5], key = 9, k = 1
+Output: [1,2,3,4,5,6]
+Explanation: Here, nums[2] == key and nums[5] == key.
+- For index 0, |0 - 2| > k and |0 - 5| > k, so there is no j where |0 - j| <= k and nums[j] == key. Thus, 0 is not a k-distant index.
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/2200-2299/2200-Find All K-Distant Indices in an Array.md:26]: replace  `>` to `&gt;`

File: dsa-solutions/lc-solutions/2200-2299/2200-Find All K-Distant Indices in an Array.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+nums = [3, 4, 9, 1, 3, 9, 5]
+key = 9
+k = 1
+print(find_k_distant_indices(nums, key, k))  # Output: [1, 2, 3, 4, 5, 6]
+
+```
+
+### Complexity Analysis
+
+**Time Complexity:** O(nlogn)
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/2200-2299/2200-Find All K-Distant Indices in an Array.md:174]: replace `O(nlogn)` to `$O(nlogn)$`

File: dsa-solutions/lc-solutions/2200-2299/2200-Find All K-Distant Indices in an Array.md [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+
+> Reason: for identifying key indices , marking k-distinct indices and for sorting
+**Space Complexity:** O(n)
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/2200-2299/2200-Find All K-Distant Indices in an Array.md:177]: replace `O(n)` to `$O(n)$`

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
Please update the code as per the suggestions and replace `<` with `&lt;` and `>` with `&gt;` because `<` and `>` are interpreted as tags.

**Comment by AmruthaPariprolu** [comment]
made required changes pls verify

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Pull request merged**